### DISCUSSION TITLE
1.4/2.0 style delete with session synchronize

### DISCUSSION THREAD

**Discussion by sebarti** [sqlalchemy/sqlalchemy#7370] [Usage Questions]
How would i delete with 1.4/2.0 style session.execute() and a delete() statement, without synchronizing the session. I need to delete many rows. I use a mapped class to create the query similar to:

```
stmt = delete(MappedClass)
session.execute(stmt)
session.commit()
```

Is this synchonized? It seemed to be very slow.

There was an option with Query where i created the delete statement like:
```
session.query(MappedClass).delete(synchronize_session=False)
session.commit()
```

**Comment by zzzeek** [comment]
new-style execute uses execution_options:

```py
stmt = delete(MappedClass)
session.execute(stmt, execution_options={"synchronize_session": False})
session.commit()
```

docs / examples at https://docs.sqlalchemy.org/en/14/orm/session_basics.html#update-and-delete-with-arbitrary-where-clause  , those docs are doing it like this:


```py
stmt = delete(MappedClass).execution_options(synchronize_session=False)
session.execute(stmt)
session.commit()
```

same thing

**Reply by zzzeek** [reply]
also to answer "is it synchronized" turn on echo=True and you'll see what it's doing
